### PR TITLE
Add stage 5 & stage 6

### PR DESCRIPTION
**Pull request by A-Quarter-Mile**
Hi @ftshijt, here are new updates on Muskit:

- Add stage 5 & stage 6.
- In stage 5, we use pitch from xml and align it according to lab_timeseq. In this version, data is collected when duration predictor is not used. 

Recipe naive_rnn without DP on Ofuton can be runned now (stage 1 ~ 5 is tested, stage 6 is still running, others need to be done)
Some details need to be checked in NAR model with DP, which hasn't been applied in this version.

- Add stage 7 (Decoding) & 8 (Scoring) & 9 (Pack model).
- Add evalute_semitone & evalute_vuv.

Recipe ofuton naive_rnn has been tested from stage 1 to stage 9.

### CODE REVIEW

File: espnet2/train/preprocessor.py [hunk 1]
```diff
@@ -856,3 +858,271 @@ def _speech_process(
...
+                        # noise: (Nmic, Time)
+                        noise = noise.T
+
+                        noise_power = (noise**2).mean()
+                        scale = (
+                            10 ** (-noise_db / 20)
+                            * np.sqrt(power)
+                            / np.sqrt(max(noise_power, 1e-10))
+                        )
+                        singing = singing + scale * noise
```
Comment (ftshijt) [inline espnet2/train/preprocessor.py:1105]: Similarly, you can remove this part

**Review by ftshijt — COMMENTED** [review]
Many thanks! A few comments

File: espnet2/fileio/xml_scp.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+from typing import Union
+
+import numpy as np
+from typeguard import check_argument_types
+import music21 as m21
+
+from espnet2.fileio.read_text import read_2column_text
+
+
+class XMLScpReader(collections.abc.Mapping):
```
[the remark this answers is not in the document]
Comment (A-Quarter-Mile) [reply espnet2/fileio/xml_scp.py:13]: The present XMLScpReader extract note info and ignore rest_note. It is designed for sub xmls after segmentation.
For XMLWriter, additional information (eg. offset, <br>, instrument, etc) might be required.

File: espnet2/svs/naive_rnn/naive_rnn.py [hunk 3]
```diff
@@ -0,0 +1,632 @@
...
+        text = text[:, : text_lengths.max()]  # for data-parallel
+        feats = feats[:, : feats_lengths.max()]  # for data-parallel
+        midi = midi[:, : midi_lengths.max()]  # for data-parallel
+        label = label[:, : label_lengths.max()]  # for data-parallel
+        batch_size = feats.size(0)
+
+        label_emb = self.encoder_input_layer(label)  # FIX ME: label Float to Int
+        midi_emb = self.midi_encoder_input_layer(midi)
+
+        if self.use_mixup_training and flag_IsValid == False:
```
[the remark this answers is not in the document]
Comment (A-Quarter-Mile) [reply espnet2/svs/naive_rnn/naive_rnn.py:354]: Removed.

File: espnet2/train/preprocessor.py (hunk 1 above)
Comment (A-Quarter-Mile) [reply espnet2/train/preprocessor.py:1105]: Removed.

File: espnet2/train/preprocessor.py [hunk 4]
```diff
@@ -856,3 +858,271 @@ def _speech_process(
...
+        g2p_type: str = None,
+        unk_symbol: str = "<unk>",
+        space_symbol: str = "<space>",
+        non_linguistic_symbols: Union[Path, str, Iterable[str]] = None,
+        delimiter: str = None,
+        rir_scp: str = None,
+        rir_apply_prob: float = 1.0,
+        noise_scp: str = None,
+        noise_apply_prob: float = 1.0,
+        noise_db_range: str = "3_10",
```
[the remark this answers is not in the document]
Comment (A-Quarter-Mile) [reply espnet2/train/preprocessor.py]: noise_scp, noise_apply_prob, noise_db_range and Add Noise have been removed.
Do we have to remove rir_scp, rir_apply_prob and codes related with rir, like Convolve RIR?

[the remark this answers is not in the document]
Comment (ftshijt) [reply espnet2/train/preprocessor.py]: Yeah, please

**Comment by ftshijt** [comment]
Thanks for the update. I will first merge it

**Pull request merged**